crates/nu-command/src/filesystem/save.rs

```diff
@@ -230,13 +241,7 @@ fn string_binary_list_value_to_bytes(value: Value, span: Span) -> Result<Vec<u8>
         }
         // Propagate errors by explicitly matching them before the final case.
         Value::Error { error } => Err(error),
-        other => Err(ShellError::OnlySupportsThisInputType(
-            "string, binary or list".into(),
-            other.get_type().to_string(),
-            span,
-            // This line requires the Value::Error match above.
-            other.expect_span(),
-        )),
+        other => Ok(other.as_string()?.into_bytes()),
```

Comment:
What are the intended semantics here? Was this a recent regression or is this making things more implicit?

Reply:
Previously, when calling `input.into_value(span)`, if the input is `PipelineData::ListStream`, the value will be `Value::List`, then value_to_bytes will try to map each element as string.  

It will goes to error if we're trying to iterate through ListStream and doesn't try to convert from element(maybe integer) to string